### ISSUE TITLE
PACS Datasplits Validation Set (To choose hyper parameters such as learn rate, best model)

### ISSUE THREAD

**Issue by NiteshBharadwaj** [mancinimassimiliano/latent_domains_DA#3]
In the paper, it is written that all the data is used for training for each domain.
So, for PAC as source and S as target, all labelled data from PAC and all unlabeled data from S should have been used for training.

In this case, on which set are hyper-parameters chosen?

**Comment by mancinimassimiliano** [comment]
Hi @NiteshBharadwaj and thanks for your interest in our work!

In our paper, we started by validating the hyperparameters for the _source only_ baseline. In that case, we performed cross-validation on the _source_ domain, randomly splitting data of that domain among train and validation (repeating the process multiple times). We used this procedure for the base set of hyperparameters (e.g., learning rate, epochs, weight decay). For the method-specific hyperparameters (i.e., weights of the entropy losses), we used the training loss curves and the probabilities given by the domain classification branch to set them. In particular, entropy loss might lead the model to diverge quickly (e.g., predicting always the same domain): we just ensured that our values were neither too high (avoiding collapsing) nor too low (to allow for confident predictions on the domain branch + learning from target data). 

Finally, I want to highlight that our setting is different w.r.t. the standard DG one of [[1]](https://arxiv.org/pdf/1710.03077.pdf), this is why we _did not_ use the standard split and validation procedures the author's proposed. Moreover, the paper _does not_ report the results of the best model(s), but as the average of multiple runs given the hyperparameters obtained as described previously.

Hope I answered your question!
